lxd/storage/drivers/driver_pure_util.go

```diff
@@ -232,3 +278,120 @@ func (p *pureClient) login() error {
+			return nil, api.StatusErrorf(http.StatusNotFound, "Storage pool %q not found", poolName)
+		}
+
+		return nil, fmt.Errorf("Failed to get storage pool %q: %w", poolName, err)
+	}
+
+	if len(resp.Items) == 0 {
+		return nil, api.StatusErrorf(http.StatusNotFound, "Storage pool %q not found", poolName)
+	}
+
+	return &resp.Items[0], nil
+}
+
+// createStoragePool creates a storage pool (PureStorage Pod).
+func (p *pureClient) createStoragePool(poolName string, size int64) error {
+	reqBody := make(map[string]any)
+	if size > 0 {
+		reqBody["quota_limit"] = size
+	}
+
+	pool, err := p.getStoragePool(poolName)
+	if err == nil && pool.IsDestroyed {
+		// Storage pool exists in destroyed state, therefore, restore it.
+		reqBody["destroyed"] = false
+
+		req, err := p.createBodyReader(reqBody)
+		if err != nil {
+			return err
+		}
+
+		err = p.requestAuthenticated(http.MethodPatch, fmt.Sprintf("/pods?names=%s", poolName), req, nil)
+		if err != nil {
+			return fmt.Errorf("Failed to restore storage pool %q: %w", poolName, err)
+		}
+
+		logger.Warn("Storage pool has been restored", logger.Ctx{"pool": poolName})
+	} else {
+		req, err := p.createBodyReader(reqBody)
+		if err != nil {
+			return err
+		}
+
+		// Storage pool does not exist in destroyed state, therefore, try to create a new one.
+		err = p.requestAuthenticated(http.MethodPost, fmt.Sprintf("/pods?names=%s", poolName), req, nil)
+		if err != nil {
+			return fmt.Errorf("Failed to create storage pool %q: %w", poolName, err)
+		}
+	}
+
+	return nil
+}
+
+// deleteStoragePool deletes a storage pool (PureStorage Pod).
+func (p *pureClient) deleteStoragePool(poolName string) error {
+	pool, err := p.getStoragePool(poolName)
+	if err != nil {
+		if api.StatusErrorCheck(err, http.StatusNotFound) {
+			// Storage pool has been already removed.
+			return nil
+		}
+
+		return err
+	}
+
+	// To delete the storage pool, we need to destroy it first by setting the destroyed property to true.
+	// In addition, we want to destroy all of its contents to allow the pool to be deleted.
+	// If the pool is already destroyed, we can skip this step.
+	if !pool.IsDestroyed {
+		req, err := p.createBodyReader(map[string]any{
+			"destroyed": true,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = p.requestAuthenticated(http.MethodPatch, fmt.Sprintf("/pods?names=%s&destroy_contents=true", poolName), req, nil)
+		if err != nil {
+			perr, ok := err.(*pureError)
+			if ok && perr.IsNotFoundError() {
+				return nil
+			}
+
+			return fmt.Errorf("Failed to destroy storage pool %q: %w", poolName, err)
+		}
+	}
+
+	// Eradicate the storage pool by permanently deleting it along all of its contents.
+	err = p.requestAuthenticated(http.MethodDelete, fmt.Sprintf("/pods?names=%s&eradicate_contents=true", poolName), nil, nil)
+	if err != nil {
+		perr, ok := err.(*pureError)
+		if ok {
+			if perr.IsNotFoundError() {
+				return nil
+			}
+
+			if perr.Matches(http.StatusBadRequest, "Cannot eradicate pod") {
```

Comment:
does pure storage have the ability to return error strings in other languages?

Or are these documented "constants"?